### PR TITLE
2864 simplify crop logic

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

**Comment by jcfr** [comment]
Cc: @ihnorton  @lassoan

File: Modules/Loadable/CropVolume/Logic/vtkSlicerCropVolumeLogic.cxx [hunk 1]
```diff
@@ -140,74 +146,42 @@ int vtkSlicerCropVolumeLogic::Apply(vtkMRMLCropVolumeParametersNode* pnode)
 
   // check the output volume type
   vtkMRMLDiffusionTensorVolumeNode *dtvnode= vtkMRMLDiffusionTensorVolumeNode::SafeDownCast(inputVolume);
+  if(dtvnode)
```
Comment (lassoan) [inline Modules/Loadable/CropVolume/Logic/vtkSlicerCropVolumeLogic.cxx]: why it is not supported?

File: Modules/Loadable/CropVolume/Logic/vtkSlicerCropVolumeLogic.cxx [hunk 2]
```diff
@@ -140,74 +146,42 @@ int vtkSlicerCropVolumeLogic::Apply(vtkMRMLCropVolumeParametersNode* pnode)
...
-
-    outputVolume = outputVVNode.GetPointer();
+    outputVolume =
+        vtkSlicerVolumesLogic::CloneVolume(scene, vvnode, outSS.str().c_str());
     }
   else if(svnode)
     {
-    outputVolume = this->Internal->VolumesLogic->CloneVolume(this->GetMRMLScene(), inputVolume, outSS.str().c_str());
+    outputVolume =
+        vtkSlicerVolumesLogic::CloneVolume(scene, inputVolume, outSS.str().c_str());
```
Comment (lassoan) [inline Modules/Loadable/CropVolume/Logic/vtkSlicerCropVolumeLogic.cxx]: Why can't you just call the CloneVolume the same way for all volume types?

File: Modules/Loadable/CropVolume/Logic/vtkSlicerCropVolumeLogic.cxx [hunk 3]
```diff
@@ -140,74 +146,42 @@ int vtkSlicerCropVolumeLogic::Apply(vtkMRMLCropVolumeParametersNode* pnode)
...
     }
   else
     {
     std::cerr << "Input volume not recognized!" << std::endl;
     return -1;
     }
 
-  outputVolume->SetName(outSS.str().c_str());
+  //outputVolume->SetName(outSS.str().c_str());
```
Comment (vovythevov) [inline Modules/Loadable/CropVolume/Logic/vtkSlicerCropVolumeLogic.cxx]: Remove commented line ?

**Comment by jcfr** [comment]
Thanks for the review.

I kept the function as it is to ensure existing code expecting `vtkMRMLScalarVolumeNode` would work.

I think I could remove the templates and have the following signatures:

```
  # expecting scalar returning scalar (for backward compat). I will check with extension uses this signature
  vtkMRMLScalarVolumeNode *CloneVolume(vtkMRMLScalarVolumeNode *volumeNode, const char *name);

  static vtkMRMLScalarVolumeNode *CloneVolumeWithoutImageData(vtkMRMLScene *scene,
                                                              vtkMRMLScalarVolumeNode *volumeNode,
                                                              const char *name);

  static vtkMRMLScalarVolumeNode *CloneVolume(vtkMRMLScene *scene,
                                              vtkMRMLScalarVolumeNode *volumeNode,
                                              const char *name,
                                              bool cloneImageData=true);

 # Generic version
  vtkMRMLVolumeNode *CloneVolume(vtkMRMLVolumeNode *volumeNode, const char *name);

  static vtkMRMLVolumeNode *CloneVolumeWithoutImageData(vtkMRMLScene *scene,
                                                              vtkMRMLVolumeNode *volumeNode,
                                                              const char *name);

  static vtkMRMLVolumeNode *CloneVolume(vtkMRMLScene *scene,
                                              vtkMRMLVolumeNode *volumeNode,
                                              const char *name,
                                              bool cloneImageData=true);
```

**Comment by jcfr** [comment]
Use of `CloneVolume`:

```
$ ack CloneVolume
SlicerProstate/DistanceMapBasedRegistration/DistanceMapBasedRegistration.py
426:    movingImageClone = volumesLogic.CloneVolume(movingImageNode,'MovingImageCopy')

ChangeTracker/ChangeTracker/ChangeTrackerWizard/ChangeTrackerAnalyzeROIStep.py
295:    followupVolumeROI = slicer.modules.volumes.logic().CloneVolume(slicer.mrmlScene, followupVolume, 'followupROI')

CarreraSlice/src/carreraslice_wrap/Documentation/GrowCutCL.py
107:      steeredVolume = volumesLogic.CloneVolume(

CarreraSlice/src/carreraslice_wrap/Documentation/VolumeMasker.py
180:      self.outputVolumeNode = volumesLogic.CloneVolume(slicer.mrmlScene, self.inputVolumeNode, outputName)

CarreraSlice/src/carreraslice_wrap/CarreraSliceEffect.py
540:    steeredVolume = volumesLogic.CloneVolume(slicer.mrmlScene, self.labelNode, steeredName)

Reporting/Py/DICOMSegmentationPlugin.py
283:      mergeNode = volumeLogic.CloneVolume(labelNode, seriesName + "-label")

Chest_Imaging_Platform/Resources/Docs/Slicer useful snippets.py
82:newvol = vl.CloneVolume(slicer.mrmlScene, node, "myNode")
138:		labelMapCopyNode = logic.CloneVolume(labelMapNode, "Copy_10270J_INSP_STD_JHU_COPD_bodyComposition") 

Chest_Imaging_Platform/Scripted/CIP_Common/CIP/logic/SlicerUtil.py
177:        # resultVolume = vlogic.CloneVolume(bigVolume, smallVolume.GetName() + "_extended")
208:        This method was implemented following the same guidelines as in slicer.modules.volumes.logic().CloneVolume(),

Chest_Imaging_Platform/Scripted/attic/PectoralisSegmentation/PectoralisSegmentation.py
318:    outputVolume = volumesLogic.CloneVolume(slicer.mrmlScene, inputVolume, 'Volume_Out')

Chest_Imaging_Platform/Scripted/CIP_InteractiveLobeSegmentation/CIP_InteractiveLobeSegmentation.py
404:                   clonedCTNode = volumesLogic.CloneVolume(slicer.mrmlScene, CTNode, 'Cloned Volume')

mpReview/mpReview.py
1828:    resampledLabelNode = slicer.modules.volumes.logic().CloneVolume(slicer.mrmlScene,labelNode,"translated")
```

**Comment by jcfr** [comment]
Closing. Superseded by [r25686](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=25686)

**Pull request closed without merging**